src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/OutputFactory.java

```diff
@@ -373,6 +397,10 @@ public Void visitVariable(Declaration.Variable tree, Declaration parent) {
             return null;
         }
 
+        if (isUnsupported(layout)) {
+            warn("skipping " + fieldName + " because of unsupported type usage: " + layout.name().get());
```

Comment:
shouldn't we have two warnUnsupported, one taking a layout, and one taking a descriptor - and have some code reuse between them?